CrossPlatformDesktopProject/CrossPlatformDesktopProject/NPC/Bat/Bat.cs

```diff
@@ -9,18 +9,49 @@ class Bat : INpc
         public INpcState currentState;
         public float xPos, yPos;
         public Rectangle hitbox;
-        public int hitboxX, hitboxY;
+        public int hitboxX, hitboxY, movementRNG;
         public Game1 myGame;
+        public System.Random random;
 
         public Bat(float xPos, float yPos, Game1 game)
         {
+            this.random = new System.Random();
             this.myGame = game;
-            currentState = new BatWalkEast(this);
             this.xPos = xPos;
             this.yPos = yPos;
             hitboxX = 16;
             hitboxY = 16;
             hitbox = new Rectangle((int)xPos, (int)yPos, hitboxX, hitboxY);
+
+            movementRNG = random.Next(1, 8);
+
+            switch(movementRNG)
```

Comment:
I think you could take the switch and put it into Update, then remove the switch from all the direction classes and rely solely on the overarching Bat.cs class to determine what the next state will be. That would be a bit cleaner and prevent the large amount of repetition between the bat classes, assuming it would work how I expect it to.

Reply:
I could've implemented it wrong, but I tried doing that and it made the NPCs very buggy. Might be because I use a counter within each directional class to determine when the next state is. So I tried putting the counter inside the bat class but yeah it didn't end up too hot. Might try and work on fixing this later more thoroughly though.